packages/discord.js/src/structures/Embed.js

```diff
@@ -211,9 +211,30 @@ class Embed {
    */
   equals(other) {
     if (other instanceof Embed) {
-      return isEqual(other.data, this.data);
+      return isEqual(this.data, other.data);
     }
-    return isEqual(other, this.data);
+
+    return (
+      this.color === (other.color ?? null) &&
+      this.description === (other.description ?? null) &&
+      this.title === (other.title ?? null) &&
+      this.url === (other.url ?? null) &&
+      (this.timestamp ? new Date(this.timestamp).getTime() : null) ===
+        (other.timestamp ? new Date(other.timestamp).getTime() : null) &&
```

Comment:
```suggestion
      this.timestamp === (other.timestamp ?? null) &&
```

Reply:
They're in different string formats so can't directly compare:
- 2024-02-26T04:52:15.531000+00:00 
- 2024-02-26T04:52:15.531Z

Reply:
```suggestion
      (this.timestamp && Date.parse(this.timestamp)) === (other.timestamp ? Date.parse(other.timestamp) : null) &&
```